src/main/java/org/prebid/server/hooks/execution/provider/ab/ABTestHookDecorator.java

```diff
@@ -0,0 +1,145 @@
+package org.prebid.server.hooks.execution.provider.ab;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.vertx.core.Future;
+import org.prebid.server.hooks.execution.v1.HookDecorator;
+import org.prebid.server.hooks.execution.v1.InvocationResultImpl;
+import org.prebid.server.hooks.execution.v1.analytics.ActivityImpl;
+import org.prebid.server.hooks.execution.v1.analytics.ResultImpl;
+import org.prebid.server.hooks.execution.v1.analytics.TagsImpl;
+import org.prebid.server.hooks.v1.Hook;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationContext;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.PayloadUpdate;
+import org.prebid.server.hooks.v1.analytics.Activity;
+import org.prebid.server.hooks.v1.analytics.Tags;
+import org.prebid.server.util.ListUtil;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class ABTestHookDecorator<PAYLOAD, CONTEXT extends InvocationContext>
```

Comment:
And then this becomes something like `ABTestedHook`, which will inform users that its actually a `decorator` and it adds `AB testing` functionality to some hook. So the code still will be written in idiomatic way.